backend/models/role.js

```diff
@@ -16,95 +16,72 @@
  */
 
 "use strict";
-(() => {
-
-	const mongoose = require("mongoose");
-	const ModelFactory = require("./factory/modelFactory");
-	const C = require("../constants");
-
-	const schema = mongoose.Schema({
-		_id : String,
-		role: String,
-		privileges: {},
-		roles: []
-	});
-
-	schema.statics.createTeamSpaceRole = function (account) {
-		const roleId = `${account}.${C.DEFAULT_MEMBER_ROLE}`;
-
-		return Role.findByRoleID(roleId).then(roleFound => {
-			if(roleFound) {
-				roleFound = roleFound.toObject();
-				return { role: roleFound.role, db: roleFound.db};
-			} else {
-				const roleName = C.DEFAULT_MEMBER_ROLE;
-				const createRoleCmd = {
-					"createRole": roleName,
-					"privileges":[{
-						"resource":{
-							"db": account,
-							"collection": "settings"
-						},
-						"actions": ["find"]}
-					],
-					"roles": []
-				};
-
-				return ModelFactory.dbManager.runCommand(account, createRoleCmd).then(()=> {
-					return {role: roleName, db: account};
-				});
-			}
-		});
 
-	};
-
-	schema.statics.dropTeamSpaceRole = function (account) {
-		const dropRoleCmd = {
-			"dropRole" : C.DEFAULT_MEMBER_ROLE
-		};
+const C = require("../constants");
+const DB = require("../handler/db");
 
-		return this.findByRoleID(`${account}.${C.DEFAULT_MEMBER_ROLE}`).then(role => {
-			if(!role) {
-				return Promise.resolve();
-			} else {
-				return ModelFactory.dbManager.runCommand(account, dropRoleCmd);
-			}
-		});
+const Role = {};
 
-	};
+Role.createTeamSpaceRole = async function (account) {
+	const roleId = `${account}.${C.DEFAULT_MEMBER_ROLE}`;
 
-	schema.statics.grantTeamSpaceRoleToUser = function (username, account) {
+	const roleFound = await Role.findByRoleID(roleId);
 
-		const grantRoleCmd = {
-			grantRolesToUser: username,
-			roles: [{role: C.DEFAULT_MEMBER_ROLE, db: account}]
-		};
+	if(roleFound) {
+		return { role: roleFound.role, db: roleFound.db};
```

Comment:
i don't think this function actually needs to return anything, it's never used (and makes sense, as it's just to make sure the db is setup.)

![image](https://user-images.githubusercontent.com/11945337/105487388-e965f880-5ca7-11eb-93be-9b5f61fa79fe.png)